utils/utils.py

```diff
@@ -643,6 +646,81 @@ def update_gdl_checkpoint(checkpoint: Union[dict, DictConfig]) -> Dict:
     return checkpoint
 
 
+def get_file_size(filepath):
+    """
+    Copied from https://instructobit.com/tutorial/117/Detecting-files-in-the-process-of-being-copied-or-written-to-in-Python#Detecting-changed-made-to-large-files-using-size-comparison
+    """
+    # open the file in read only
+    with open(filepath, "r") as file:
+        # move pointer to the end of the file
+        file.seek(0, 2)
+        # retrieve the current position of the pointer
+        # this will be the file's size in bytes
+        size = file.tell()
+        return size
+    # if the function reaches this statement it means an error occurred within the above context handler
+    return False
+
+
+def wait_while_modif(fpath: Union[str, Path], sleep_secs: int = 10, timeout: int = 1800) -> None:
```

Comment:
did you have unit tests for these new functions?

Reply:
Easy answer: yes and no.
Yes: I added a test `test_download_url_wcheck(self):` under test_utils.py that call this function.
No: no test for this function directly. I have a hard time figuring out how to test this. I'd have to find a way to modify a file in background while checking if it's being modified with this function. I find the test mentionned above sufficient...